vault/resource_approle_auth_backend_role_secret_id.go

```diff
@@ -388,3 +415,7 @@ func approleAuthBackendRoleSecretIDParseID(id string) (backend, role, accessor s
 
 	return
 }
+
+func isAppRoleDoesNotExistError(err error, role string) bool {
+	return util.Is500(err) && strings.Contains(err.Error(), fmt.Sprintf("role \"%s\" does not exist", role))
```

Comment:
Nice! I was thinking this week about these types of scenarios and we probably don't always handle them correctly like this.